### PR TITLE
chore: Update PR template

### PR DESCRIPTION
**Pull request by ivadym**
## Description

[PR description]

#### Added

- [Description]

#### Changed

- [Description]

#### Fixed

- [Description]

#### Removed

- [Description]

## Documentation

- [ ] N/A
- [ ] Updated Balsamic documentation to reflect the changes as needed for this PR.
    - [Document Name]

### Feature Tests

- [ ] N/A
- [ ] Test [Description]
    - [Screenshot]

### Pipeline Integrity Tests

- **Report deliver (generation of the `.hk` file)**
    - [x] N/A
    - [ ] Verified
- **TGA T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **TGA T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **QC Workflow**
    - [x] N/A
    - [ ] Verified
- **PON Workflow**
    - [x] N/A
    - [ ] Verified

## Clinical Genomics Stockholm

<details>
<summary></summary>

### Documentation

- **Atlas documentation**
    - [x] N/A
    - [ ] Updated: [Link]
- **Web portal for Clinical Genomics**
    - [x] N/A
    - [ ] Updated: [Link]

### User Changes

- [x] N/A
- [x] This PR affects the output files or results.
    - [ ] User feedback is considered unnecessary because [Justification].
    - [ ] Affected users have been included in the development process and given a chance to provide feedback.

### Infrastructure Changes

- **Stored files in Housekeeper**
    - [x] N/A
    - [ ] Updated: [Link]
- **CG (CLI and delivered/uploaded files)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Servers (configuration files on Hasta)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Scout interface**
    - [x] N/A
    - [ ] Updated: [Link]

</details>

## Checklist

> [!IMPORTANT]  
> Ensure that all checkboxes below are ticked before merging.

### For Developers

- **PR Description**
    - [ ] Provided a comprehensive description of the PR.
    - [ ] Linked relevant user stories or issues to the PR.
- **Documentation**
    - [ ] Verified and updated documentation if necessary.
- **Tests**
    - [ ] Described and tested the functionality addressed in the PR.
    - [ ] Ensured integration of the new code with existing workflows.
    - [ ] Confirmed that meaningful unit tests were added for the changes introduced.
    - [ ] Checked that the PR has successfully passed all relevant code smells and coverage checks.
- **Review**
    - [ ] Addressed and resolved all the feedback provided during the code review process.
    - [ ] Obtained final approval from designated reviewers.

### For Reviewers

- **Code**
    - [ ] Code implements the intended features or fixes the reported issue.
    - [ ] Code follows the project's coding standards and style guide.
- **Documentation**
    - [ ] Pipeline changes are well-documented in the CHANGELOG and relevant documentation.
- **Tests**
    - [ ] The author provided a description of their manual testing, including consideration of edge cases and boundary
      conditions where applicable, with satisfactory results.
- **Review**
    - [ ] Confirmed that the developer has addressed all the comments during the code review.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,17 +1,59 @@
...
+<!-- Provide a brief overview of your PR. -->
+<!-- Link any relevant user stories or issues. -->
+<!-- Ensure that all checkboxes below are ticked before merging. -->
+
+## Infrastructure Changes
+
+<!-- Link related issues or PRs for necessary changes. -->
+
+- [ ] Updated delivered and uploaded files (Hermes) as needed for this PR.
+- [ ] Ensured CG interface and CLI are updated to align with the changes.
```
Comment (fevac) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: and cg.config

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,17 +1,59 @@
...
+
+<!-- Link all added or updated documents. -->
+
+- [ ] Updated Balsamic documentation to reflect the changes as needed for this PR.
+- [ ] Ensured Atlas documentation is updated accordingly.
+
+## Tests
+
+<!-- Describe in detail how you tested your changes to help reviewers validate the code. -->
+<!-- Include screenshots or visual representations of your changes. -->
```
Comment (fevac) [inline .github/PULL_REQUEST_TEMPLATE.md:25]: I would add something like: Never disclose private information, and if relevant add it in the internal folder

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,17 +1,59 @@
...
+- [ ] Test B
+- [ ] Not applicable
+
+## Checklist for Reviewers
+
+- [ ] Code follows the project's coding standards and style guide.
+- [ ] Code implements the intended features or fixes the reported issue.
+- [ ] Code is accompanied by relevant unit tests.
+- [ ] Pipeline changes are well-documented in the CHANGELOG and relevant documentation.
+- [ ] Manual testing performed by the author is described; they considered edge cases and boundary conditions, and the results are satisfactory.
```
Comment (fevac) [inline .github/PULL_REQUEST_TEMPLATE.md:57]:
```suggestion
- [ ] Manual testing performed by the author is described; when relevant they consider edge cases and boundary conditions, and the results are satisfactory.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,17 +1,59 @@
...
-(If this is a release PR, no need to add following. Leave this part empty)
-(Use the following lines to create a PR text body. Make sure to remove all non-relevant one after you're done)
-(Repeat each field as many times as necessary)
-
-Added: for new features.
-Changed: for changes in existing functionality.
-Deprecated: for soon-to-be removed features.
-Removed: for now removed features.
-Fixed: for any bug fixes.
-Security: in case of vulnerabilities.
```
Comment (fevac) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: I think we should keep this (added/changed/fixed/etc)

**Review by fevac — APPROVED** [review]
Nice!

**Review by fevac — COMMENTED** [review]
Although thinking about it I'm not sure if this should be here or in the MTP repo. Many of these things are irrelevant for Balsamic and only relevant for CG

**Comment by ivadym** [comment]
> Although thinking about it I'm not sure if this should be here or in the MTP repo. Many of these things are irrelevant for Balsamic and only relevant for CG

Yes, I know. I'm also not so happy about adding this here, but we need a place to document properly the tests carried out and the tools involved. I also thought about moving it to the issues, but then we'll end up again merging things that have not been properly integrated with our interfaces. Maybe we can bring this up for a refinement session next week.

Thanks for the review!

**Comment by fevac** [comment]
> Maybe we can bring this up for a refinement session next week.

Let's do that. I would vote for making MTP-Balsamic private and adding all these things here but let's decide next week

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,17 +1,59 @@
...
+
+- [ ] Updated Balsamic documentation to reflect the changes as needed for this PR.
+- [ ] Ensured Atlas documentation is updated accordingly.
+
+## Tests
+
+<!-- Describe in detail how you tested your changes to help reviewers validate the code. -->
+<!-- Include screenshots or visual representations of your changes. -->
+
+### Feature Tests
```
Comment (mathiasbio) [inline .github/PULL_REQUEST_TEMPLATE.md:27]: I wonder if it would be possible to create some type of criteria like for the validation reports. Something that forces the developer to consider the impacts of their changes on the pipeline, and how to set requirements that need to pass for the PR to be allowed to be merged.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,17 +1,59 @@
...
-Fixed: for any bug fixes.
-Security: in case of vulnerabilities.
-
-### Review and tests: 
-- [ ] Tests pass
-- [ ] Code review
-- [ ] New code is executed and covered by tests, and test approve
+## Description
+
+<!-- Provide a brief overview of your PR. -->
```
Comment (mathiasbio) [inline .github/PULL_REQUEST_TEMPLATE.md:3]: I wonder if it may be required to add some checkbox here as well. Like 
- [x] Description of PR has been given
Otherwise there may be a risk that it's ignored

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (pbiology) [reply .github/PULL_REQUEST_TEMPLATE.md:27]: It's a good idea. It tricky to think of a template that fits many scenarios. Maybe it is also the responsibility of the reviewer to ensure that there a well described tests so one can understand if the developer have thought it through?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -1,17 +1,59 @@
...
+<!-- Link any relevant user stories or issues. -->
+<!-- Ensure that all checkboxes below are ticked before merging. -->
```
Comment (mathiasbio) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: Maybe here as well, and for the user stories. If it would be possible to convert the comments to checkboxes. I feel like adding checkboxes makes it feel less optional 🤔

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (mathiasbio) [reply .github/PULL_REQUEST_TEMPLATE.md:27]: Maybe something like a small risk-assessment, with a checkbox that needs to be checked. 
- [x] Risks have been assessed and considered in tests

**Review by mathiasbio — APPROVED** [review]
I love this, thanks a lot! 🌟

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (ivadym) [reply .github/PULL_REQUEST_TEMPLATE.md:3]: Yes, I think it's a good idea :)

**Comment by pbiology** [comment]
> > Maybe we can bring this up for a refinement session next week.
> 
> Let's do that. I would vote for making MTP-Balsamic private and adding all these things here but let's decide next week

Yeah let's discuss it. I'm not sure how to do with PRs in the MTP repo since we won't have any code there

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (ivadym) [reply .github/PULL_REQUEST_TEMPLATE.md:12]: In my opinion it ends up being a copy paste of the CHANGELOG. Maybe it's good if we make reviewers go and check throughly what has been added there

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -1,17 +1,59 @@
...
+<!-- Include screenshots or visual representations of your changes. -->
+
+### Feature Tests
+
+<!-- Include tests relevant to the changes in this PR. -->
+
+- [ ] Test A
+- [ ] Test B
+- [ ] Not applicable
+
```
Comment (mathiasbio) [inline .github/PULL_REQUEST_TEMPLATE.md:34]: I have another potential section we could add, but it could change quite a lot the way we work. But I think it could be nice to add some section like:

Customer relevant changes:
- [x] change significantly affects customer results
If above is checked:
- [x] affected customers / users have been included in the development process and given a chance to provide feedback

Comment (ivadym) [reply .github/PULL_REQUEST_TEMPLATE.md:34]: I like it 🔥

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (pbiology) [reply .github/PULL_REQUEST_TEMPLATE.md:27]: Very good point. Kinda spaced on that.

Awesome!

**Comment by pbiology** [comment]
So just a general comment, it is possible to have multiple PR templates, but it requires browser extensions to make it easy, like this one:  
[https://chromewebstore.google.com/detail/github-pr-templates/cpebkokcapgjobabnjpnnndbgbjddool](https://chromewebstore.google.com/detail/github-pr-templates/cpebkokcapgjobabnjpnnndbgbjddool)  
  
Not saying we should do it, but just an FYI that these things exist

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (ivadym) [reply .github/PULL_REQUEST_TEMPLATE.md:12]: I've added it again, was using this template and it's kind of nice to list the changes

**Review by mathiasbio — APPROVED** [review]
This looks great to me! Well done! 🌟 Looking forward to using it and seeing it be used :D Fantastic!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -1,17 +1,132 @@
...
+#### Pipeline Integrity Tests
+
+<!-- Include tests to verify the integrity of the different Balsamic workflows. -->
+
+- [ ] N/A
+- [ ] Verified integrity of the Balsamic TGA T/O workflow.
+- [ ] Verified integrity of the Balsamic TGA T/N workflow.
+- [ ] Verified integrity of the Balsamic WGS T/O workflow.
+- [ ] Verified integrity of the Balsamic WGS T/N workflow.
+- [ ] Verified integrity of the Balsamic QC workflow.
```
Comment (mathiasbio) [inline .github/PULL_REQUEST_TEMPLATE.md:49]: Though I missed this part, I think we should add integrity for the UMI-workflows too. And I also wonder if we should let the "N/A" optionality be on a workflow-basis. Sometimes a change only affects one workflow for instance and it would be nice if we could skip some if it's unnecessary to test them : )

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 10]
```diff
@@ -1,17 +1,132 @@
...
+- [ ] Updated _Atlas_ documentation for this PR.
+  - [Link]
+- [ ] Ensured web portal for Clinical Genomics reflects changes.
+  - [Link]
+
+### User Changes
+
+- [ ] N/A
+- [ ] Affected users have been included in the development process and given a chance to provide feedback.
+
```
Comment (mathiasbio) [inline .github/PULL_REQUEST_TEMPLATE.md:73]: I wonder if we could add a text-box here where we could encourage and explanation if we put N/A. In some cases the "N/A" will be obvious, but in others it can be a gray-zone, like changes may affect the results but we have ensured that the results are improvements and feedback is then unnecessary. But for instance especially in the development of new features, feedback is more important

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 9 above)
Comment (ivadym) [reply .github/PULL_REQUEST_TEMPLATE.md]: 👍 [7e6ca81](https://github.com/Clinical-Genomics/BALSAMIC/pull/1369/commits/7e6ca810816781ab887482769fa510cc9189023a)

Comment (ivadym) [reply .github/PULL_REQUEST_TEMPLATE.md]: Do you think we need to differentiate between T/O & T/N? 🤔

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 10 above)
Comment (ivadym) [reply .github/PULL_REQUEST_TEMPLATE.md:94]: Got your point! I've updated it but feel free to rephrase it or restructure it: [8e353c9](https://github.com/Clinical-Genomics/BALSAMIC/pull/1369/commits/8e353c9b7a151c3ee31f28ddeef1419ee0f419ab)

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 9 above)
Comment (mathiasbio) [reply .github/PULL_REQUEST_TEMPLATE.md]: I think so yes! But I don't know how to format it to avoid it being messy 🤔

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 11]
```diff
@@ -1,17 +1,165 @@
...
+    - [ ] N/A
+    - [ ] Verified
+- **PON Workflow**
+    - [ ] N/A
+    - [ ] Verified
+
+## Clinical Genomics Stockholm
+
+<details>
+<summary></summary>
```
Comment (ivadym) [inline .github/PULL_REQUEST_TEMPLATE.md:79]: I've been testing the template in my PRs, and it seems that when you click a checkbox, it closes the collapsible content. It is a known issue: https://github.com/orgs/community/discussions/42523.

I was wondering if we should mark `N/A` by default for this section so it's not so annoying 🙃

Comment (fevac) [reply .github/PULL_REQUEST_TEMPLATE.md:79]: Yes, marking NA would be nice then

**Review by fevac — APPROVED** [review]
Nice! ⭐ Very comprehensive but with the check boxes hopefully doesn't become too annoying 😬

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 12]
```diff
@@ -1,17 +1,165 @@
...
+    - [ ] N/A
+    - [ ] Updated: [Link]
+- **Servers (configuration files on Hasta)**
+    - [ ] N/A
+    - [ ] Updated: [Link]
+- **Scout interface**
+    - [ ] N/A
+    - [ ] Updated: [Link]
+
+### Integration Tests
```
Comment (mathiasbio) [inline .github/PULL_REQUEST_TEMPLATE.md:120]: Like we discussed on slack, this may be unnecessary as these integration tests are done in their respective PRs on Hermes / CG etc! : )
